### PR TITLE
Added the ShareCard screen 

### LINKED ISSUE #12: Image Cropper
- Circular crop 
- Rectangular crop (set aspect ratio)

### PR DESCRIPTION
**Pull request by Abishek1997**
ref #13 

- Added the ShareCard screen which contains data, composables and Components 
- Exposes and uses the ShareCardCustom() and ShareCardDefault() reusable composable UI functions
- Material3 theme styled components are added
- Added GlideImage composable to load images and drawables
- Further requirements are handled in drawables, component's gradle, Manifest file to setup the ShareCard
- Comments are added throughout the file explaining the usage of each composable/ function

**In portrait:**
**Light vs Dark themes**

    

**In landscape:**
**Light vs Dark themes**

Screenshot_20221014_120001

Screenshot_20221014_115902

### CODE REVIEW

File: app/src/main/java/com/compose/playground/MainActivity.kt [hunk 1]
```diff
@@ -74,6 +70,9 @@ fun Navigation() {
         composable("Timeline Intro") {
             TimelineIntro(navController = navController)
         }
+        composable("Share Card"){
```
Comment (siddeshpillai) [inline app/src/main/java/com/compose/playground/MainActivity.kt:73]: @Abishek1997 missing space `){`

File: components/build.gradle [hunk 2]
```diff
@@ -44,7 +44,17 @@ dependencies {
...
     androidTestImplementation 'androidx.test.espresso:espresso-core:3.4.0'
+
+    // Glide-compose
+    implementation "com.github.bumptech.glide:compose:1.0.0-alpha.0"
+
+    // Composable screenshot
+    implementation 'com.github.SmartToolFactory:Compose-Screenshot:1.0.0'
+
+    // Accompanist
+    implementation "com.google.accompanist:accompanist-drawablepainter:0.26.4-beta" // drawablePainter
```
Comment (siddeshpillai) [inline components/build.gradle:59]: @Abishek1997 Why do you need this?

File: components/src/main/res/drawable/ic_baseline_more_horiz_24.xml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+<vector android:autoMirrored="true" android:height="24dp"
+    android:tint="#000000" android:viewportHeight="24"
```
Comment (siddeshpillai) [inline components/src/main/res/drawable/ic_baseline_more_horiz_24.xml:2]: @Abishek1997 Can you use the vector provided by the SDK?

File: components/src/main/res/drawable/ic_baseline_share_24.xml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+<vector android:autoMirrored="true" android:height="24dp"
+    android:tint="#000000" android:viewportHeight="24"
+    android:viewportWidth="24" android:width="24dp" xmlns:android="http://schemas.android.com/apk/res/android">
```
Comment (siddeshpillai) [inline components/src/main/res/drawable/ic_baseline_share_24.xml:3]: @Abishek1997 Can you use the vector provided by the SDK?

**Review by SidoPillai — CHANGES_REQUESTED** [review]
@Abishek1997 Some minor comments to resolve.

File: components/src/main/java/com/compose/components/ShareCard.kt [hunk 5]
```diff
@@ -0,0 +1,505 @@
...
+        )
+    )
+}
+
+/**
+ * The wrapper page to display the ShareScreen() composable with a bottom scaffold and provides the required state variables
+ */
+@Composable
+@Stable
+fun ShareScreenScaffold(
```
Comment (siddeshpillai) [inline components/src/main/java/com/compose/components/ShareCard.kt:133]: `()` can be in one line since there are no args

**Comment by Abishek1997** [comment]
Made the above mentioned changes.

**Review by SidoPillai — CHANGES_REQUESTED** [review]
If possible rely primary and onPrimary colors. 
Image is not circular for icon
text and background of the shade card needs to designed for both light and dark theme
image

Switching theme caused results in image loading issues 
image

**Comment by Abishek1997** [comment]
Made changes to use Material3 surface, primary, onSurface, onPrimary variants for all UI based elements. Replaced the screenshots in the original comment

**Review by SidoPillai — APPROVED** [review]

**Pull request merged**